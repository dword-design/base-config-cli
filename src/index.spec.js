import { Base } from '@dword-design/base'
import { execaCommand } from 'execa'
import fs from 'fs-extra'
import withLocalTmpDir from 'with-local-tmp-dir'

export default {
  async afterEach() {
    await this.resetWithLocalTmpDir()
  },
  async beforeEach() {
    this.resetWithLocalTmpDir = await withLocalTmpDir()
  },
  private: async () => {
    await fs.outputFile('package.json', JSON.stringify({ private: true }))
    await new Base({ name: '../src/index.js' }).prepare()
    expect(await fs.exists('ecosystem.json')).toBe(false)
  },
  async works() {
    await execaCommand('git init')
    await execaCommand(
      'git remote add origin git@github.com:dword-design/foo.git',
    )
    await fs.outputFile('package.json', JSON.stringify({ name: 'foo' }))
    await new Base({ name: '../src/index.js' }).prepare()
    expect(await fs.readJson('ecosystem.json')).toMatchSnapshot(this)
  },
}
