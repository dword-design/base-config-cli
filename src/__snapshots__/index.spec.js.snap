// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index works 1`] = `
Object {
  "deploy": Object {
    "production": Object {
      "host": Array [
        "sebastianlandwehr.com",
      ],
      "path": "/var/www/foo",
      "post-deploy": "source ~/.nvm/nvm.sh && yarn --frozen-lockfile && yarn prepublishOnly",
      "ref": "origin/master",
      "repo": "git@github.com:dword-design/foo.git",
      "user": "root",
    },
  },
}
`;
